util/execdetails/execdetails.go

```diff
@@ -141,6 +161,46 @@ func (e *RuntimeStatsColl) Get(planID string) *RuntimeStats {
 	return runtimeStats
 }
 
+// RecordOneCopTask records a specific cop tasks's execution detail.
+func (e *RuntimeStatsColl) RecordOneCopTask(planID, address string,
+	timeProcessedNs, numProducedRows, numIterations uint64) {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+
+	_, ok := e.copStats[planID]
+	if !ok {
+		e.copStats[planID] = make(map[string][]*CopRuntimeStats, 8)
+	}
+
+	e.copStats[planID][address] = append(e.copStats[planID][address],
+		&CopRuntimeStats{timeProcessedNs, numProducedRows, numIterations})
+}
+
+// CopSummary gets a summary of the cop task's execution information.
+func (e *RuntimeStatsColl) CopSummary(planID string) string {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+	stats, ok := e.copStats[planID]
+	if !ok {
+		return ""
+	}
+	var totalRows, totalIters, totalTasks uint64
+	procTimes := make([]uint64, 0, 32)
+	for _, instanceStats := range stats {
+		for _, stat := range instanceStats {
```

Comment:
Seems there is no need to differentiate the instance?

Reply:
It's not necessary now, but for further use, distinguishing the instance is useful.
For example, to trace which TiKV instance is the bottleneck of a query, we may list all tasks for each instance.